src/compile-gate.lisp

```diff
@@ -155,7 +157,49 @@ QUBITS should be a NAT-TUPLE of qubits representing the Hilbert space."
                  operator
                  amps
                  arefs))))
-           :dotimes-iterator 'lparallel:pdotimes)))))
+           :dotimes-iterator **dotimes-iterator**)))))
+
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;; MEASUREMENTS ;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(declaim (ftype (function (bit bit) bit) bit=)
+         (inline bit=))
+(defun bit= (a b)
+  (ldb (byte 1 0) (logeqv a b))
+  #+#:other (- 1 (logxor a b))
+  #+#:other (if (= a b) 1 0))
+
+(defun generate-single-qubit-measurement-code (qubit &key (dotimes-iterator **dotimes-iterator**))
```

Comment:
For my education, could you write a paragraph or so about this function? what/why/how it does what it do